src/analysis/convex/basic.lean

```diff
@@ -444,6 +444,26 @@ lemma convex.sub {t : set E}  (hs : convex s) (ht : convex t) :
   convex ((λx : E × E, x.1 - x.2) '' (s.prod t)) :=
 (hs.prod ht).is_linear_image is_linear_map.is_linear_map_sub
 
+-- This is also true when p and/or q are zero
+lemma convex.smul_add_smul (h_conv : convex s) {p q : ℝ} (hp : 0 < p) (hq : 0 < q) :
```

Comment:
Then can you replace `<` with `\le`?

Reply:
I could but it makes the proof longer with little practical benefit as anyone who has 0 smul some set should likely be simping that away to the set `{0}`, the comment is more of a curiosity.

Reply:
Sure, but if the caller has only `0 ≤ p` you're forcing them to do the case split rather than just being able to use this lemma.

Reply:
True of course, I would hope that that future user would instead just generalize this lemma if they found a use for it. If you really want I can add it, I'm just explaining why I didn't bother so far.